### PR TITLE
HPCC-28516 Report no Log Access Plug-in when accessing workunit log

### PR DESCRIPTION
**Pull request by wangkx**
When Log Access plug-in not available, send a warning for: 1. ZAP Report dialog; 2. log display request.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by GordonSmith — APPROVED** [review]

**Review by afishbeck — APPROVED** [review]
@wangkx looks fine

**Comment by wangkx** [comment]
@ghalliday this PR has been rebased to the latest master. It is ready to be merged.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 1]
```diff
@@ -323,10 +323,31 @@ void WsWuInfo::sendImportedWorkunitComponentLog(const char* logFile, CHttpRespon
     response->send();
 }
 
+void WsWuInfo::sendErrorMessage(CHttpResponse* response, LogAccessLogFormat format, const char* message)
+{
+    if (format == LOGACCESS_LOGFORMAT_json)
+    {
+        StringBuffer resp;
+        resp.set("{ ");
+        resp.appendf("\"Error\": \"%s\"", message);
```
Comment (ghalliday) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp:332]: minor: If this is a general purpose function then it should be json-encoding the text (e.g. quoting double quotes) 
Should this be a member of CHttpResponse?  Is it a general function for an http response, or specific?

**Review by ghalliday — APPROVED** [review]
@wangkx Happy to merge as-is, but do you want to address the encoding question?

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (wangkx) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp:332]: I am going to use the appendJSONStringValue function to get the json right. Also create a CHttpResponse.setErrorLogContent().

**Comment by wangkx** [comment]
@ghalliday please review last 2 commits.

**Review by ghalliday — APPROVED** [review]
@wangkx please squash

**Comment by wangkx** [comment]
@ghalliday I just did squash.

**Pull request merged**